configuration.c

```diff
@@ -1511,6 +1511,23 @@ static struct config_array_setting *populate_settings_array(
    SETTING_ARRAY("input_android_physical_keyboard", settings->arrays.input_android_physical_keyboard, false, NULL, true);
 #endif
 
+   SETTING_ARRAY("input_player1_reserved_device",  settings->arrays.input_player1_reserved_device,  false, NULL, true);
```

Comment:
Same here, dehardcode all this.

Reply:
I was able to dehardcode everything besides these calls to `SETTING_ARRAY`, I couldn't figure out a way to get having the key dynamically generated in a loop to work, but it's probably because my C is rusty: https://github.com/libretro/RetroArch/pull/16269/commits/2aeeb237c9783a09b1902f9aa99336bf89163bbc

I noticed another player-specific setting being hardcoded so I ended up doing the same thing: https://github.com/libretro/RetroArch/blob/adaa19cbf7f5d1ad7847ab0156e052e35a20f665/configuration.c#L2141-L2156